src/components/CountryPickerMenuItem.js

```diff
@@ -0,0 +1,66 @@
+import React, {useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import styles from '../styles/styles';
+import Navigation from '../libs/Navigation/Navigation';
+import ROUTES from '../ROUTES';
+import useLocalize from '../hooks/useLocalize';
+import MenuItemWithTopDescription from './MenuItemWithTopDescription';
+import FormHelpMessage from './FormHelpMessage';
+
+const propTypes = {
+    /** Error text from form, e.g when no country is selected */
+    errorText: PropTypes.string,
+    /** function from form to call when the country changes, important for revalidation */
+    onInputChange: PropTypes.func.isRequired,
+    /** Prop which states when country value changed */
+    didCountryChange: PropTypes.bool.isRequired,
+    /** Function for setting didCountryChange to false  */
+    setDidCountryChange: PropTypes.func.isRequired,
```

Comment:
That would cause infinite rerender. When you call `onInputChange` inside `CountryPickerMenuItem`, the `Form` will create a new `CountryPickerMenuItem`, which will call `onInputChange` again inside the `useEffect` and so on.

Reply:
Something like below?
```
useEffect(()=>{
  onInputChange(route.params.country)
},[route.params.country])
```

But as i said 
> That would cause infinite rerender. When you call onInputChange inside CountryPickerMenuItem, the Form will create a new CountryPickerMenuItem, which will call onInputChange again inside the useEffect and so on.

So, you need to store when the route changed in the parent component, and hence the below 
```
useEffect(() => {
    // This will cause the form to revalidate and remove any error related to country name
    if (!didCountryChange) return;
    onInputChange(countryCode);
    setDidCountryChange(false);
}, [didCountryChange, onInputChange, setDidCountryChange, countryCode]);
```

Reply:
Maybe in the future when the `Form` component's behaviour is modified, we could have alternate solutions